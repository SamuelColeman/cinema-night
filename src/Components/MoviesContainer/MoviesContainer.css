/* font-family: 'Staatliches', cursive;
font-family: 'Fjalla One', sans-serif;
font-family: 'Anton', sans-serif; */

.movies_container {
  display: flex;
  flex-wrap: wrap;
  height: 100vh;
  overflow: scroll;
  justify-content: center;
}

.div__movie--header {
  display: flex;
  flex-direction: column;
  width: 100%;
  justify-content: space-around; 
  align-items: center;
}

.img--login {
  max-width: 17%;
  max-height: 13vh;
}

.img--logo {
  max-width: 17%;
}

.title-link,
.title {
  font-family: 'Anton', sans-serif;
  margin: 4%;
  letter-spacing: 5px;
  margin: 2%;
}

/* .h4--login {
  margin: 2%;
} */

.title {
  font-size: 1.5em;
}

.title-link {
  font-size: 1.5em;
  text-decoration: none;
  color: black;
  
}

.h1-cinema-night {
  font-family: 'Limelight', cursive;
  margin-bottom: 0;
  font-size: 4em;
  margin-top: 2%;
}

/* header {
  display: flex;
  flex-direction: column;
  justify-content: center;
} */

.div__movie--title {
  background-image: url(/static/media/board.2890075f.png);
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  background-size: cover;
  width: 32%;
}

.h4--login {
  margin: 0;
}

