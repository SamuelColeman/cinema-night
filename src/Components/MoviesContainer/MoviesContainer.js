import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import './MoviesContainer.css';
import { NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const MoviesContainer = ({ currentUser, movies, signOutUser, selectMovie }) => {
  let button;
  if (currentUser.name === undefined || currentUser.id === null) {
    button = (
      <NavLink className="title-link" to='/login'>
        <h4 className="h4--login" onClick={signOutUser}>Sign In</h4>
        {/* <button>Sign In</button> */}
      </NavLink>
    )
  } else {
    button = (
      <NavLink className="title-link" to='/login'>
        <h4 className="h4--login" onClick={signOutUser}>Sign Out</h4>
        {/* <button onClick={signOutUser}>Sign Out</button> */}
      </NavLink>
    )
  }
  const loopMovies = movies.map((movie) => {
      return <MovieCard key={movie.id}
              selectMovie={selectMovie}
                        {...movie} />
  })
    return (
        <section className='movies-containers'>
          <header>
          <h1 className="h1-cinema-night">CINEMA NIGHT</h1>
          <div className="div__movie--title">
          <h6 className="h6--playing title">Now Playing</h6>
          <Link to='/favorites'>
            <h4 className="h4--view-favorites title-link">View Favorites</h4>
          </Link>
        {button}

        </div>
          </header>
          <div className="movies">
          {loopMovies}
          </div>
        </section>
    ) 
} 

export const mapStateToProps = (state) => ({
  currentUser: state.currentUser,
  movies: state.movies
})

export default connect(mapStateToProps)(MoviesContainer);