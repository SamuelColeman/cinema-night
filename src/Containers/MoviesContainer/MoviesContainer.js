import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import './MoviesContainer.css';
import { NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const MoviesContainer = ({ currentUser, movies, signOutUser, handleFavourite }) => {
    let signINbutton = (
      <NavLink className="title-link" to='/login'>
        <h4 className="h4--login" onClick={signOutUser}>Sign In</h4>
        {/* <button>Sign In</button> */}
      </NavLink>
    )
    let signOUTbutton = (
      <NavLink className="title-link" to='/login'>
        <h4 className="h4--login" onClick={signOutUser}>Sign Out</h4>
      </NavLink>
    )
  const loopMovies = movies.map((movie) => {
      return <MovieCard key={movie.id}
              handleFavourite={handleFavourite}
                        {...movie} />
  })
    return (
        <section className='movies-containers'>
          <header>
          <h1 className="h1-cinema-night">CINEMA NIGHT</h1>
          <div className="div__movie--title">
          <h6 className="h6--playing title">Now Playing</h6>
          <Link to='/favorites'>
            <h4 className="h4--view-favorites title-link">View Favorites</h4>
          </Link>
        { currentUser.name === undefined || currentUser.id === null ? signINbutton : signOUTbutton }
        </div>
          </header>
          <div className="movies">
          {loopMovies}
          </div>
        </section>
    ) 
} 

export const mapStateToProps = (state) => ({
  currentUser: state.currentUser,
  movies: state.movies
})

export default connect(mapStateToProps)(MoviesContainer);