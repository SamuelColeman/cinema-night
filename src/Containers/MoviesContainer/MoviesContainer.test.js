 import React from 'react';
 import { shallow } from 'enzyme';
 import MoviesContainer from './MoviesContainer';


 describe('MoviesContainer',  () => {
    let wrapper;
    const mockMovies = [
        {
          id: 456,
          title: 'Harry Potter and the Sorceres Stone',
          popularity: 462.91,
          vote_count: 3775,
          video: false,
        },
        {
            id: 123,
            title: 'Harry Potter and the Sorceres Stone',
            popularity: 462.91,
            vote_count: 3775,
            video: false,
          }
    ]

    beforeEach(() => {
      wrapper = shallow( <MoviesContainer 
                movies = {mockMovies}
                errorMsg= 'Error'
      />)
    })

    it('should match snapshot with all data being passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
 })