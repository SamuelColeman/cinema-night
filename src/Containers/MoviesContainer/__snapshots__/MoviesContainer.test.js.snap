// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainerOfContainers MoviesContainer should match snapshot with all data being passed in correctly 1`] = `
<section
  className="movies-containers"
>
  <header>
    <h1
      className="h1-cinema-night"
    >
      CINEMA NIGHT
    </h1>
    <div
      className="div__movie--title"
    >
      <h6
        className="h6--playing title"
      >
        Now Playing
      </h6>
      <Link
        to="/favorites"
      >
        <h4
          className="h4--view-favorites title-link"
        >
          View Favorites
        </h4>
      </Link>
      <NavLink
        className="title-link"
        to="/login"
      >
        <h4
          className="h4--login"
          onClick={[MockFunction]}
        >
          Sign Out
        </h4>
      </NavLink>
    </div>
  </header>
  <div
    className="movies"
  >
    <Connect(MovieCard)
      active="inactive"
      handleFavourite={[MockFunction]}
      id={456}
      key="456"
      overview="yada yada"
      popularity={462.91}
      poster_path="https://pottersworld.com"
      release_date="10-02-10"
      title="Harry Potter and the Sorceres Stone"
      vote_avarage={3775}
    />
    <Connect(MovieCard)
      active="inactive"
      handleFavourite={[MockFunction]}
      id={457}
      key="457"
      overview="yada yada"
      popularity={462.91}
      poster_path="https://pottersworld.com"
      release_date="10-02-10"
      title="Harry Potter and the Sorceres Stone"
      vote_avarage={3775}
    />
  </div>
</section>
`;
